apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/GoogleDiagnosticsStartupFilter.cs

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Google.Cloud.Logging.V2;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Google.Cloud.Diagnostics.AspNetCore
+{
+    /// <summary>
+    /// A startup filter to configure Google Diagnostics services.
+    /// </summary>
+    internal class GoogleDiagnosticsStartupFilter : IStartupFilter
+    {
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly IServiceProvider _serviceProvider;
+        private readonly string _projectId;
+        private readonly LoggerOptions _loggerOptions;
+        private readonly LoggingServiceV2Client _loggingServiceClient;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="GoogleDiagnosticsStartupFilter"/> class.
+        /// </summary>
+        /// <param name="loggerFactory">The <see cref="ILoggerFactory"/> to configure logging.</param>
+        /// <param name="serviceProvider">
+        /// The <see cref="IServiceProvider"/> to pass to resolve additional services.
+        /// </param>
+        /// <param name="projectId">
+        /// The Google Cloud Platform project ID. If unspecified and running on GAE or GCE
+        /// the project ID will be detected from the platform.
+        /// </param>
+        /// <param name="loggerOptions">Optional, options for the logger.</param>
+        /// <param name="loggingServiceClient">Optional, logging client.</param>
+        public GoogleDiagnosticsStartupFilter(
```

Comment:
I've added a bunch of optional parameters to specify options for several components.

Do you think they are necessary? We could also say that if you need more control you have to resort to calling the individual methods (the current way). Personally these options don't bother me for most apps, I just want them to have good defaults.

Reply:
The more I think about it the more I prefer this option (but I think we should keep project id here).  This seems reasonable as most people (ideally) won't need to change the defaults.